### ISSUE TITLE
AspNetCoreHostingModel InProcess

### ISSUE THREAD

**Issue by cgountanis** [serilog/serilog-aspnetcore#75]
How do we go about configuring Serilog now that AspNetCoreHostingModel INProcess breaks the .SetBasePath(Directory.GetCurrentDirectory()). Will there be installation help documentation and sample udpates soon? Should we fall back to configuring in Startup.cs?

csproj
` <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>`

**Comment by adamchester** [comment]
Hi @cgountanis, firstly, I’m really not sure, I haven’t tried it yet, but I intend to soon. I’ll try to ask some questions that might help move this forward, please forgive me if I fail 🙃

1. Can you elaborate on how `SetBasePath` (From Microsoft.Extensions.Configuration?) is involved? I assume you’re loading from app settings, a json file in the directory of the main application? Would [AppContext.BaseDirectory](
https://docs.microsoft.com/en-us/dotnet/api/system.appcontext.basedirectory?view=netcore-2.2) help? 
2. I can guess how the `CurrentDirectory` would be different or have an unexpected value when using a different hosting model. In fact, I faced this issue even prior to the new InProcess model.

**Comment by Hawxy** [comment]
> Do you know what it’s set to for this in process model?

You'd probably be interested in reading through this: https://github.com/aspnet/AspNetCore/issues/4206

TL;DR, this [helper ](https://github.com/aspnet/Docs/blob/master/aspnetcore/host-and-deploy/aspnet-core-module/samples_snapshot/2.x/CurrentDirectoryHelpers.cs) is now required to set the correct directory until .NET Core 3.0 is out

**Comment by cgountanis** [comment]
Just wondering if it would be easier to configure it in startup. There does not seem to be any examples 4 for 2.2 startup configuration.

**Comment by Hawxy** [comment]
Supposedly you can just call that helper method at startup and the rest of your project will function the same as before. Probably the best option since you can easily get rid of it with the next version of .net core.

**Comment by leomwa** [comment]
This tripped me up last night. I solved it by not calling the SetBasePath method __and__ removing the static initializer by converting the property into a method.
This is the code that works for me at the moment.

```
public class Program
{
    private static IConfiguration GetConfig()
    {
        return new ConfigurationBuilder()
            .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
            .Build();
        // skipped the aditional/optional appsettings.<env>.json file for brevity
    }

    public static int Main(string[] args)
    {
        Log.Logger = new LoggerConfiguration()
            .ReadFrom.Configuration(GetConfig()) // notice the call to GetConfig here
            .Enrich.FromLogContext()
            .CreateLogger();  

    // ... etc
    }

    // ... create webhost builder
}
```

I could be wrong, but my theory is that at the point of initialisation (before Main is called) the SetBasePath + Directory.GetCurrentDirectory() argument [of the property initializer](https://github.com/serilog/serilog-aspnetcore/blob/dev/samples/SimpleWebSample/Program.cs#L12) sets the paths [as documented](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-2.2#in-process-hosting-model) but if we do not call SetBasePath it is [inferred from ApplicationBase](https://github.com/aspnet/Announcements/issues/88). 🤷‍♂️

**Comment by cgountanis** [comment]
Thanks for all the help, going to wait for Core 3.0 since there might be some changes and just tackle the InProcess thing at that point to save some time.

**Comment by nblumhardt** [comment]
Closing as it seems like this is a configuration issue rather than directly related to this library, but when we ship the next version I'll try to get an updated sample app into the repository.
